src/main/java/com/box/sdk/BoxFile.java

```diff
@@ -758,6 +758,42 @@ public void deleteMetadata(String typeName, String scope) {
         request.send();
     }
 
+    /**
+     * Used to retrieve the watermark for the file.
+     * If the file does not have a watermark applied to it, a 404 Not Found will be returned by API.
+     * @param fields the fields to retrieve.
+     * @return the watermark associated with the file.
+     */
+    public BoxWatermark getWatermark(String... fields) {
+        return this.getWatermark(FILE_URL_TEMPLATE, fields);
+    }
+
+    /**
+     * Used to apply or update the watermark for the file.
+     * @return the watermark associated with the file.
+     */
+    public BoxWatermark applyWatermark() {
+        return this.applyWatermark(FILE_URL_TEMPLATE, BoxWatermark.WATERMARK_DEFAULT_IMPRINT);
+    }
+
+    /**
+     * Removes a watermark from the file.
+     * If the file did not have a watermark applied to it, a 404 Not Found will be returned by API.
+     */
+    public void removeWatermark() {
+        this.removeWatermark(FILE_URL_TEMPLATE);
+    }
+
+    private String scopeBasedOnType(String typeName) {
```

Comment:
Please remove this method.  BoxFile has been refactored since you created this branch. and this method has been moved to the MetaData class.